### PR TITLE
iaito: 5.7.6 -> 5.7.8

### PR DESCRIPTION
**Pull request by azahi**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/iaito/default.nix [hunk 1]
```diff
@@ -9,60 +9,77 @@
...
+      rev = "ab923335409fa298c39f0014588d78d926c6f3a2";
+      hash = "sha256-qkIC67a6YRwOa2Sr16Vg6If1TmAiSKUV7hw13Wxwl/w=";
+      name = repo;
+    })
+  ];
+  sourceRoot = "iaito/src";
 
   postUnpack = ''
-    sourceRoot=$sourceRoot/src
+    chmod -R u+w -- iaito-translations
```
Comment (SuperSandro2000) [inline pkgs/tools/security/iaito/default.nix:37]:
```suggestion
    chmod -R u+w iaito-translations
```
FYI we do not need that here

File: pkgs/tools/security/iaito/default.nix [hunk 2]
```diff
@@ -9,60 +9,77 @@
...
+  ];
+
+  buildInputs = [
+    qtbase
+    radare2
+  ];
+
+  postBuild = ''
+    pushd ../../../iaito-translations
+    make build PREFIX=$out
```
Comment (SuperSandro2000) [inline pkgs/tools/security/iaito/default.nix:56]:
```suggestion
    make build PREFIX=$out -j$NIX_BUILD_CORES
```

**Comment by azahi** [comment]
@SuperSandro2000 Ping.

File: pkgs/tools/security/iaito/default.nix [hunk 3]
```diff
@@ -9,60 +9,77 @@
...
-    install -m644 -Dt $out/share/metainfo $src/src/org.radare.iaito.appdata.xml
-    install -m644 -Dt $out/share/applications $src/src/org.radare.iaito.desktop
-    install -m644 -Dt $out/share/pixmaps $src/src/img/iaito-o.svg
+    install -m644 -Dt $out/share/metainfo ../org.radare.iaito.appdata.xml
+    install -m644 -Dt $out/share/applications ../org.radare.iaito.desktop
+    install -m644 -Dt $out/share/pixmaps ../img/iaito-o.svg
+
+    pushd ../../../iaito-translations
+    make install PREFIX=$out -j$NIX_BUILD_CORES
+    popd
```
Comment (symphorien) [inline pkgs/tools/security/iaito/default.nix:70]: you install translations into $out/share/iaito/translations but it looks like iaito does not loof for them there: 
```
Loading translations path /home/symphorien/.local/share/iaito/translations
Loading translations path /nix/store/qrd2nnpw3rj0ll9xkr06vzs1j87xibsn-qtbase-5.15.7/translations
```
and as a matter of fact on first startup only english is available

Comment (azahi) [reply pkgs/tools/security/iaito/default.nix:75]: Should work now. I guess it loaded them from `~/.local/share/iaito/translations` which I had installed when I was testing it out.

**Comment by azahi** [comment]
It's important to backport this package now because it supposed to follow the version of `radare2` which is already at 5.7.8 in 22.11.

File: pkgs/tools/security/iaito/default.nix (hunk 3 above)
Comment (symphorien) [reply pkgs/tools/security/iaito/default.nix:75]: I still get the same: 
```
Loading translations path /home/symphorien/.local/share/iaito/translations
Loading translations path /nix/store/qrd2nnpw3rj0ll9xkr06vzs1j87xibsn-qtbase-5.15.7/translations
```
I see you replaced the path to translations in ResourcePath.cpp but it's not event mentioned in this debug output...

Comment (azahi) [reply pkgs/tools/security/iaito/default.nix:75]: That's weird. I also get this message, but translations can now be selected from the drop down menu in the settings window (I double checked that I don't have them in `~/.local/share/iaito/translations`).

image

Comment (symphorien) [reply pkgs/tools/security/iaito/default.nix:75]: I see this drop-down, it is populated as your screenshot show, but chosing french does not seem to have any effect, even after a restart.

Comment (azahi) [reply pkgs/tools/security/iaito/default.nix:75]: I'll look into this and make another PR soon.

**Pull request merged**